mloop/learners.py

```diff
@@ -1031,6 +1035,30 @@ def _rand2(self, index):
         '''
         r0, r1, r2, r3, r4 = self.random_index_sample(index, 5)
         return self.population[r0] + self.curr_scale * (self.population[r1] + self.population[r2] - self.population[r3] - self.population[r4])
+    
+    # custom evolutionary strategy
+    # may need to add a separate scaling factor
+    def _currtobest1(self, index):
+        '''
+        Use current parameter, best parameters and two others to generate mutation.
+        (see https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.differential_evolution.html for a definition of the strategy).
+
+        Args:
+            index (int): Index of member to mutate.
+        '''   
+        r0, r1 = self.random_index_sample(index, 2)
+        return self.population[index] + self.curr_scale * (self.population[self.min_index] - self.population[index] + self.population[r0] - self.population[r1])
```

Comment:
I took a look at scipy's docs, and they cite "A Unified Differential Evolution Algorithm for Global Optimization" by Qiang and Mitchell. This and the other new method agree with equations 5 and 6 in that paper respectively, with the constraint that `$F_{cr} = F_{xc}$`. In other words, this implementation looks correct to me!